### DISCUSSION TITLE
My latest results using private dataset trained tacotron2 model with melgan vocoder

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#149] [Archived]
<USER>>>> nmstoker<br/> <DATE> [February 26, 2020, 1:45pm]<br/> <br/> I'm near the end of my lunch break, so I need to be brief and I'll add<br/> more on some skipped details this evening, but wanted to share output<br/> I've produced with my private dataset trained for Tacotron2 and also<br/> with the MelGAN vocoder.<br/> <br/> I'm really pleased with the results - whilst I can hear aspects I'd like<br/> to improve still (tone and sometimes speed), I suspect they are largely<br/> down to my dataset.<br/> <br/> For some reason the Soundcloud link isn't working here if pasted alone,<br/> but I think this should work now:<br/> <https://soundcloud.com/user-726556259/sets/tacotron2-plus-melgan-v1><br/> <br/> The output was produced by slightly adapting server.py / synthesizer.py<br/> based on the CoLab here:<br/> <https://github.com/mozilla/TTS/issues/345#issuecomment-586778834><br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/my-latest-results-using-private-dataset-trained-tacotron2-model-with-melgan-vocoder]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [February 28, 2020, 12:05pm]<br/> <br/> Nice work! which melgan implementation is that ?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [February 28, 2020, 12:06pm]<br/> <br/> Could you also share some Dataset examples to compare the results?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> nmstoker<br/> <DATE> [February 28, 2020, 1:40pm]<br/> <br/> I'll get back more fully over the weekend but I based my config on this<br/> one:<br/> <https://github.com/erogol/ParallelWaveGAN/blob/tts/parallel_wavegan/melgan_v3.yaml><br/> <br/> As I'm using a Ti1080 (ie 11Gb) I had previously had memory issues (when<br/> I'd tried training with an earlier version of this repo) so I'd slightly<br/> reduced the value for batch_max_steps. In retrospect I realise now I<br/> didn't test whether those issues came up still, I just assumed they<br/> might.<br/> <br/> Happy to share some samples from the dataset (which I'll also have to do<br/> over the weekend<br/> Are you after examples of the full length recordings used to train TTS<br/> or the samples that ParallelWaveGAN produces (ie those shorter snippets<br/> it includes along with the generated audio output during training)<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [February 28, 2020, 1:41pm]<br/> <br/> I was more after the ground-truth audio samples of your dataset you used<br/> to train your models. Thx for sharing all this btw.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> nmstoker<br/> <DATE> [March 1, 2020, 1:22pm]<br/> <br/> Here's some more detail, including some audio samples used for training.<br/> <br/> **My audio dataset** *slash* <br/> The samples here are just picked randomly from the full set. Overall the<br/> full set varies in sentence length reasonably well. They have been<br/> recorded over quite some time, so unfortunately in my wider set there<br/> are likely to be some with slightly inconsistent style but I've<br/> generally tried to keep it neutral and steady, weeding out some of the<br/> most different / worst cases by use of the Speaker Embedding chart and a<br/> bit of brute effort<br/> <br/> The dataset is getting on for 16+ hours now. I'm happy to share the<br/> handful of samples included here, but wish to keep my overall dataset<br/> private (as it's my own voice).<br/> <br/> **Most recent training run** *slash* <br/> Having seen how good your results were with MelGAN here:<br/> <https://github.com/mozilla/TTS/issues/345> *slash* <br/> I planned to follow as closely as I could the same basic approach simply<br/> using my dataset instead:<br/> <br/> - First using Forward Attention enabled until *slash* ~400K (*although<br/> ended up being 525k in the end*)<br/> - Then finetuning with BN<br/> [here](https://github.com/erogol/ParallelWaveGAN/tree/tts/parallel_wavegan)<br/> <br/> One small difference is that I'm using ng-espeak, with UK RP phonemes<br/> (given that's closer to my accent than the default US English output)<br/> and it's a slightly customised version as I recompiled the dictionary<br/> for it with some missing words which feature fairly commonly in my<br/> dataset (which the default ng-espeak UK RP setting was not converting<br/> accurately). I suspect the impact of the customisation is fairly small<br/> for training.<br/> <br/> When I ran the initial run for Tacotron2 (see 1_config.json in zip) I<br/> let it run for longer and ended up using the best model from around 525k<br/> as the one that I then finetuned with BN (2_config.json)<br/> <br/> Training the vocoder was fairly straight forward, although initially I<br/> got a little mixed up about the file metadata.txt coming from<br/> **bin/preprocess_tts.py** (I may be wrong, but I think in the earlier<br/> version it was using csv format train and validate metadata files and I<br/> initially tried with those files, as I'd used them when I looked at this<br/> repo late last year; anyway no big deal - the file format was rejected<br/> and I soon figured it out - will teach me to read the code more before<br/> jumping in!!<br/> <br/> The vocoder ran for the full 400k times. The config for that is<br/> **melgan_v3_neil15.yaml** in the zip file attached. I changed the value<br/> for **batch_max_steps** as mentioned above, but otherwise it was based<br/> on<br/> [melgan_v3.yaml](https://github.com/erogol/ParallelWaveGAN/blob/tts/parallel_wavegan/melgan_v3.yaml)<br/> <br/> I did wonder if perhaps I should train MelGAN even longer for better<br/> results - that's just a matter of setting the **resume** argument isn't<br/> it?<br/> <br/> Hope that helps - fire away with any other questions etc. I'll be back<br/> online again this evening.<br/> <br/> [Original_audio_samples_and_config_files.zip](https://discourse.mozilla.org/uploads/short-url/tPEsCMWbO1AJ1hyhsHMBaBKFMk1.zip)<br/> (468.4 KB)<br/> <br/> [](https://discourse.mozilla.org/t/train-multispeaker-dataset-wavernn/52012/21)<br/> <br/> [Train Multispeaker Dataset + WaveRNN<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> nmstoker<br/> <DATE> [March 2, 2020, 12:51am]<br/> <br/> Am now trying a longer run with MelGAN. Resume parameter seems to be<br/> working fine, although it did look like it was stuck at first (just took<br/> a long time with no obvious change after showing the run parameters, but<br/> then the progress bar reappeared with 400k, going up to my new 800k<br/> target)<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [March 2, 2020, 12:19pm]<br/> <br/> Thx for sharing all these.<br/> <br/> Regarding MelGAN, yes training more always help. Even after 2m<br/> iterations, you still see improvements.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> nmstoker<br/> <DATE> [March 2, 2020, 1:03pm]<br/> <br/> Good to know! Thanks.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> petertsengruihon<br/> <DATE> [March 20, 2020, 3:57am]<br/> <br/> Hey nmstoker, I have tried<br/> to trained ParallelWaveGAN on Indonesia datasets. The sample below sound<br/> not quite clear. Have you ever encountered similar situation ? Thanks a<br/> lot.<br/> <br/> [Indonesian_wavernn_sample.zip](https://discourse.mozilla.org/uploads/short-url/oqV9SWloIEtTydgZJYT8Q6I76p3.zip)<br/> (861.4 KB)<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> dkreutz<br/> <DATE> [July 14, 2020, 1:22pm]<br/> <br/> I trying MelGAN as well<br/> and I am unsure about the dataset preparation, moreover spllitting the<br/> dataset: *slash* <br/> How did you split your dataset (train + dev/eval)?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [July 14, 2020, 1:53pm]<br/> <br/> In PWGan I extract features for all wavs and then move some of them to a<br/> separate folder for dev/eval.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> dkreutz<br/> <DATE> [July 14, 2020, 2:00pm]<br/> <br/> Is there any recommendation for a split ratio, e.g. 10:1 or are there<br/> only a few feature-files for dev/eval necessary (currently I have<br/> approx. 22.000 for train and 200 for eval)?<br/> <br/> Or to put it in another way: does the dev/eval set influence the<br/> learning (I did not fully grasp the discriminator training yet)?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [July 14, 2020, 2:03pm]<br/> <br/> I think that was also my split<br/> I got good results on my single speaker, but when I tried multispeaker<br/> it was very bad. I have no idea why, but there was a lot of static. In<br/> single speaker, 400K steps were enough to get good quality.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> nmstoker<br/> <DATE> [July 14, 2020, 2:53pm]<br/> <br/> I've been putting all my audio into one set, both for ParallelWaveGAN<br/> and the recently integrated vocoder (which has just finished a first<br/> long run).<br/> <br/> Originally with ParallelWaveGAN, I'd done this to follow the training<br/> [config](https://github.com/erogol/ParallelWaveGAN/blob/tts/parallel_wavegan/melgan_v4.yml#L9):<br/> <br/> ` meta_file_train: 'metadata.txt',`<br/> <br/> and it seemed to work well there. I do see general discussions on GANs<br/> suggesting that it's still worth having test so I might try it in time.<br/> <br/> One minor point with the integrated vocoder: it just reads the wavs in a<br/> directory (rather than looking in a metadata file to see which are to be<br/> used). With both my normal training set and test set, I have a fair<br/> number of wavs present that aren't in my metadata files (I had to<br/> discard fair number of samples, I would cut the line from metadata but<br/> not remove the wav). Therefore I'd need a little tidy up and so as a<br/> shortcut I left all the audio - the main issue behind me cutting wavs<br/> from the training/test sets was transcription errors or prematurely cut<br/> recordings, neither of which should particularly impact training the<br/> MelGAN (given that it's trying to recreate the audio input; in fact it<br/> may benefit because it then has a bit more audio to learn from).<br/> <br/> FYI: I'm going to post details on my DDC runs soon (later today I hope,<br/> as I'm off work and on a '*staycation*' so for once I have time during<br/> the week!) Then will follow up later with details from the integrated<br/> vocoder (which I need to do more runs with)<br/> <br/> [Archived Post]<br/>
